### PR TITLE
ADR013: Configuration management system

### PR DESCRIPTION
**Pull request by shonfeder**
This ADR proposes adoption of the `PureConfig` library for configuration
management. Please see the ADR for details on the proposal.

Towards #1069

- [x] Ran `make fmt-fix` (or had formatting run automatically on all files edited)
- [x] Documentation added for any new functionality

### CODE REVIEW

File: docs/src/adr/013adr-configuration.md [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+  serverPort: Port = Port(),
+  writeIntermediate: Boolean = false,
+  profiling: Boolean = false,
+  outDir: Path = Path("."),
+  smtEncding: SmtEncoding = Oopsla19Encoding,
+)
+
+case classs ExampleUseOfConfigs() = {
+  val cli = CliParseResults()
+  val localConfig = ConfigSource.file(Path.cwd.resolve(".aplache.config"))
```
Comment (konnov) [inline docs/src/adr/013adr-configuration.md:200]:
```suggestion
  val localConfig = ConfigSource.file(Path.cwd.resolve(".apalache.config"))
```

File: docs/src/adr/NNNadr-template.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
+---
+authors: Octavia Buttler, Eduardo Galleano
+last revised: YYYY-MM-DD
+---
+
+# ADR-{NNN}: {Decision Subject}
```
Comment (konnov) [inline docs/src/adr/NNNadr-template.md:6]: Why is this template part included in the PR?

**Review by konnov — APPROVED** [review]
Yes! It's a great idea to replace our ad-hoc configuration with a library. PureConfig looks good.

File: docs/src/adr/NNNadr-template.md (hunk 2 above)
Comment (shonfeder) [reply docs/src/adr/NNNadr-template.md:6]: I just based this off of the template PR, so I need to rebase.

**Comment by shonfeder** [comment]
Thanks for the fix!

**Pull request merged**